### DISCUSSION TITLE
How to layout html content into columns or pageview

### DISCUSSION THREAD

**Discussion by winterdl** [daohoangson/flutter_widget_from_html#613] [Q&A]
I want to render large html content into PageView, how to measuring width height of a html content. Thank you.

**Comment by daohoangson** [comment]
It's just a normal widget so you can measure it with something like `MeasureSize` from https://github.com/flutter/flutter/issues/14488#issuecomment-719963492.

If you want to split the contents into pages, e.g. for a book reader like #138. It will be tricky. Somehow you need to break down the long HTML into smaller pieces then measure each of them before grouping a few into pages.

**Reply by winterdl** [reply]
Thank you for your reply, but I wondering if the widget have an event onRenderElement or something will call when each tag was rendered, so that we could break the content into pages if the size exceeds screen size. I'm going dev an ebook reader using webview but flutter_webview is a platform view, so that the webview could not rendering from offscreen. My ebook content contains only text, image and sometime audio, it will be good if using your package to render the book content, for now I could pagination text content using TextPainter, with getPositionForOffset method.

**Reply by getBoolean** [reply]
@winterdl were you able to find a solution for rendering paginated html? I am currently working on a custom epub library, so I am interested in how this could be done.

For reference, J-Novel Club has implemented a basic paged html reader in their Flutter mobile app, and the text is selectable. Judging from their open source libraries list, they did not use one of the html packages on pub.dev. This gives me hope it can be done without a webview.
